### PR TITLE
stm32h723Xg SoC Support

### PR DESCRIPTION
**Pull request by Nukersson**
Dear all. I know, that this is not a primer OpenOCD dev repo. But I can't manage commiting my changes to original repor. Thereofre I am putting it here. This changes working for me with my nucle_h723zg board an allows me running compiled Zephyr RTOS Hello World sample.

How to compile:
```
./bootstrap
./configure --enable-shared --disable-internal-jimtcl
make
sudo make install
```

additionaly you may need to remove compiled files:
```
make clean
```

@galak Maybe my changes could help you adding h723 support to original OpenOCD repo. I give you also my permisssion for use this code usage in original OpenOCD repo under your name.

P.S.: I am not familiar with OpenOCD development procedure and wondering why this PR changes so much files. Originally I've made changes only to `src/flash/nor/stm32h7x.c` file upon the current master branch!

### CODE REVIEW

File: src/flash/nor/stm32h7x.c [hunk 1]
```diff
@@ -804,6 +823,16 @@ static int stm32x_probe(struct flash_bank *bank)
...
+		/* For STM32H72x/73x
+		 *  - STM32H7xxxG devices contains single bank, 1024 Kbyte
+		 */
+		if (flash_size_in_kb == 128)
+			has_dual_bank = false;
+		else
+			/* flash size is 2M or 1M */
+			flash_size_in_kb /= 1;
+		break;
 	default:
```
Comment (Nukersson) [inline src/flash/nor/stm32h7x.c:836]: @galak This were my changes only! All the other changes are strange for me currently and comes from current master!

**Comment by galak** [comment]
can you make your changes against the zephyr-20200722 branch

**Comment by Nukersson** [comment]
> can you make your changes against the zephyr-20200722 branch

Done

File: src/flash/nor/stm32h7x.c (hunk 1 above)
Comment (galak) [reply src/flash/nor/stm32h7x.c:836]: Is there an upstream gerrit reference?

File: src/flash/nor/stm32h7x.c [hunk 2]
```diff
@@ -824,6 +824,14 @@ static int stm32x_probe(struct flash_bank *bank)
...
 		break;
 	case 0x483:
+		/* For STM32H72x/73x
+		 *  - STM32H7xxxG devices contains single bank, 1024 Kbyte
+		 */
+		if (flash_size_in_kb == 128)
+			has_dual_bank = false;
+		else
+			/* flash size is 2M or 1M */
+			flash_size_in_kb /= 1;
```
Comment (galak) [inline src/flash/nor/stm32h7x.c:834]: what's the point of the else?  divide `/1` seems unnecessary.

Comment (Nukersson) [reply src/flash/nor/stm32h7x.c]: You're right. My copy-paste mistake. Changed now.

**Comment by galak** [comment]
@erwango do you mind take a look at this.

**Review by erwango — CHANGES_REQUESTED** [review]
This SoC has been provided official support on openocd upstream (http://openocd.zylin.com/#/c/5792/: v0.10.0-1410-g35e580373).
So I'd prefer we use official change

**Review by erwango — CHANGES_REQUESTED** [review]
This SoC has been provided official support on openocd upstream (http://openocd.zylin.com/#/c/5792/: v0.10.0-1410-g35e580373).
So I'd prefer we use official change

**Comment by galak** [comment]
> This SoC has been provided official support on openocd upstream (http://openocd.zylin.com/#/c/5792/: v0.10.0-1410-g35e580373).
> So I'd prefer we use official change

Thanks for pointing that out.  I agree, we should pull this in from upstream.

**Comment by Nukersson** [comment]
@galak Do we need more activity here? Otherwise this PR can be closed.

**Comment by Nukersson** [comment]
Solved within OpenOCD v11

**Pull request closed without merging**